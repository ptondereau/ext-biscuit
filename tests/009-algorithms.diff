001- bool(true)
002- bool(true)
003- bool(true)
004- bool(true)
005- bool(true)
006- bool(true)
007- bool(true)
008- bool(true)
009- bool(true)
010- bool(true)
011- bool(true)
012- bool(true)
001+ Fatal error: Uncaught BiscuitException: Invalid key data length in /home/ptondereau/Code/ext-biscuit/tests/009-algorithms.php:3
002+ Stack trace:
003+ #0 /home/ptondereau/Code/ext-biscuit/tests/009-algorithms.php(3): BiscuitKeyPair->__construct()
004+ #1 {main}
005+   thrown in /home/ptondereau/Code/ext-biscuit/tests/009-algorithms.php on line 3
