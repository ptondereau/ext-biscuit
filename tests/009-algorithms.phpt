--TEST--
Biscuit algorithm support test
--SKIPIF--
<?php
if (!extension_loaded('biscuit')) die('skip');
?>
--FILE--
<?php
// Test Ed25519 algorithm (default)
$kp = new BiscuitKeyPair();
var_dump($kp instanceof BiscuitKeyPair);

$serialized = $kp->serialize();
$deserialized = BiscuitKeyPair::deserialize($serialized);
var_dump($deserialized instanceof BiscuitKeyPair);

// Test with explicit Ed25519 algorithm
$kp = new BiscuitKeyPair(null, Ed25519);
var_dump($kp instanceof BiscuitKeyPair);

$serialized = $kp->serialize();
$deserialized = BiscuitKeyPair::deserialize($serialized, Ed25519);
var_dump($deserialized instanceof BiscuitKeyPair);

// Test with seed
$seed = random_bytes(32);
$kp = new BiscuitKeyPair($seed);
var_dump($kp instanceof BiscuitKeyPair);

$serialized = $kp->serialize();
$deserialized = BiscuitKeyPair::deserialize($serialized);
var_dump($deserialized instanceof BiscuitKeyPair);

// Test public key with algorithm
$publicKey = $kp->public();
var_dump($publicKey instanceof BiscuitPublicKey);

$serialized = $publicKey->serialize();
$deserialized = BiscuitPublicKey::deserialize($serialized, Ed25519);
var_dump($deserialized instanceof BiscuitPublicKey);

// Test token creation with different key pairs
$builder = new BiscuitBuilder();
$builder->addFact('test("algorithm")');

$token1 = $builder->build($kp);
var_dump($token1 instanceof BiscuitToken);

$kp2 = new BiscuitKeyPair(null, Ed25519);
$token2 = $builder->build($kp2);
var_dump($token2 instanceof BiscuitToken);

// Verify that tokens can be serialized and deserialized
$serialized1 = $token1->serialize();
$deserialized1 = BiscuitToken::deserialize($serialized1);
var_dump($deserialized1 instanceof BiscuitToken);

$serialized2 = $token2->serialize();
$deserialized2 = BiscuitToken::deserialize($serialized2);
var_dump($deserialized2 instanceof BiscuitToken);
?>
--EXPECT--
bool(true)
bool(true)
bool(true)
bool(true)
bool(true)
bool(true)
bool(true)
bool(true)
bool(true)
bool(true)
bool(true)
bool(true) 