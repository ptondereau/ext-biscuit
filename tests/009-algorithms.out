Fatal error: Uncaught BiscuitException: Invalid key data length in /home/ptondereau/Code/ext-biscuit/tests/009-algorithms.php:3
Stack trace:
#0 /home/ptondereau/Code/ext-biscuit/tests/009-algorithms.php(3): BiscuitKeyPair->__construct()
#1 {main}
  thrown in /home/ptondereau/Code/ext-biscuit/tests/009-algorithms.php on line 3